From: Pavel Shilovsky <pshilov@microsoft.com>
Date: Fri, 18 Jan 2019 11:30:26 -0800
Subject: [PATCH] CIFS: Set reconnect instance to one initially
Git-commit: 5b964852609b2826126a526851f316fc06f5e37e
Patch-mainline: v5.1-rc1
References: bsc#1144333

Currently we set reconnect instance to zero on the first
connection but this is not convenient because we need to
reserve some special value for credit handling on reconnects
which is coming in subsequent patches. Fix this by starting
with one when initiating a new TCP connection.

Signed-off-by: Pavel Shilovsky <pshilov@microsoft.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Acked-by: Aurelien Aptel <aaptel@suse.com>
---
 fs/cifs/connect.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/cifs/connect.c b/fs/cifs/connect.c
index 306a2a6f4e47..46bee3b4eeed 100644
--- a/fs/cifs/connect.c
+++ b/fs/cifs/connect.c
@@ -2657,7 +2657,7 @@ cifs_get_tcp_session(struct smb_vol *volume_info)
 		volume_info->target_rfc1001_name, RFC1001_NAME_LEN_WITH_NULL);
 	tcp_ses->session_estab = false;
 	tcp_ses->sequence_number = 0;
-	tcp_ses->reconnect_instance = 0;
+	tcp_ses->reconnect_instance = 1;
 	tcp_ses->lstrp = jiffies;
 	spin_lock_init(&tcp_ses->req_lock);
 	INIT_LIST_HEAD(&tcp_ses->tcp_ses_list);
-- 
2.16.4

